### DISCUSSION TITLE
How to save changes to a local Realm on disk?

### DISCUSSION THREAD

**Discussion by Jean-RobertLaffineur** [realm/realm-dotnet#3647] [General]
When using and properly closing a local Realm (as described in the docs), the Realm is saved, with all the changes, to the .realm file on my Windows machine.

The problem is that when the realm is not properly closed (crash...), the changes are lost.

Is there a way to force the write on the file to persist changes at will?

Thanks.

**Comment by nirinchev** [comment]
When a `.Write` call completes, the changes are persisted to disk. You don't need to properly close the Realm file.

**Comment by Jean-RobertLaffineur** [comment]
Thank you for your answer. My previous tests were probably affected by opened transactions, thus the changes were never saved.
